src/main/java/core/basesyntax/dao/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,196 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.exceptions.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO `cars`"
+                + " (manufacturer_id, model)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection
+                         .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, car.getManufacturer().getId());
+            preparedStatement.setString(2, car.getModel());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create car " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
```

Comment:
When you get car in method get(), getAll(), getAllByDriver() you should get list of drivers and manufacturer too. HINT: you can write inner join query to get car and manufacturer from one request to DB. And then you can write separate query to get all drivers for each car.

Reply:
I get all of them, I created private methods for this in the end of the class: getDrivers and getManufacturer